File: Invert Binary Tree/Invert Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        return self.invert(root)
    
    def invert(self, node):
        if (node is not None):
            left = self.invert(node.left)
            right = self.invert(node.right)
            node.left = right
            node.right = left
        return node
            
        

File: Invert Binary Tree/Invert Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        return invert(root);
    }
    
    public TreeNode invert(TreeNode node) {
        if (node != null) {
            TreeNode left = invert(node.left);
            TreeNode right = invert(node.right);
            
            node.left = right;
            node.right = left;
        }
        return node;
    }
}